src/Pact/Native/Keysets.hs

```diff
@@ -82,6 +89,43 @@ defineKeyset g0 fi as = case as of
             runSysOnly $ enforceKeySet i (Just ksn) oldKs
             writeRow i Write KeySets ksn ks & success "Keyset defined"
 
+descKeySet :: GasRNativeFun e
+descKeySet g i [TLitString t] = do
+  r <- readRow (_faInfo i) KeySets (KeySetName t)
+  case r of
+    Just v -> computeGas' g i (GPostRead (ReadKeySet (KeySetName t) v)) $
+              return $ toTerm v
+    Nothing -> evalError' i $ "Keyset not found: " <> pretty t
+descKeySet _ i as = argsError i as
+
+descGuard :: GasRNativeFun e
+descGuard gas i [TGuard g _] = do
+  case g of
+    GKeySet ks ->
+      computeGas' gas i (GUserApp Defun) $
+      return $ toTObject TyAny def
+        [ ("type", tStr "KeySet")
```

Comment:
How should we handle type-specific metadata? I'm guessing we should only have `type` be common and everything else varies, but for some that's  JSON no-no. What about a `details` field which is a varying object? Or is that overthinking it

Reply:
The main motivating purpose for this was providing information to apps / wallets / etc about what keys they should sign with.  So the big thing my immediate use case wants from this is public keys.  Beyond that I really don't know.  Probably the most important criteria is that any changes we make down the road can be done in a backwards-compatible manner.

Reply:
I guess one other high level question is whether we should put the case-specific data at the same level or if we should make some kind of generic object wrapper to make it easier to decode.

Reply:
> case-specific data at the same level or if we should make some kind of generic object wrapper to make it easier to decode
That was what I was trying to suggest with `details`, let's do that.